src/requirementslib/models/setup_info.py

```diff
@@ -68,10 +68,7 @@
     from pip._vendor.pkg_resources import DistInfoDistribution, EggInfoDistribution
     from pip._vendor.requests import Session
 
-    try:
-        from setuptools.dist import Distribution
-    except ImportError:
-        from distutils.core import Distribution
+    from setuptools.dist import Distribution
```

Comment:
Just noting that importing anything from setuptools will cause problems being able to re-install or install a different version of setuptools.   The only reason this code is OK is because it is shielded behind `if MYPY_RUNNING:` and won't actually run operationally.   CC @oz123 this is one of the examples I was referring to earlier.

Reply:
When setuptools 60+ is insatlled, the distutils import will import from setuptools anyway.

Reply:
Well it still matters on 60+.   Ref: https://github.com/sarugaku/requirementslib/pull/339
and there are some issues that were closed in the pipenv backlog that required this change.

Reply:
Should I revert this part of the change?